test/test_reductions.py

```diff
@@ -1387,10 +1387,12 @@ def test_dim_reduction(self, device):
                  torch.int64,
                  torch.int32,
                  torch.int16]
-        if self.device_type == 'cuda':  # 'cpu' and 'xla' do not support half
+        if self.device_type == 'cuda':  # 'cpu' and 'xla' do not completely support half
```

Comment:
> Is it still true that cpu does not completely support half? 

`prod` of `float16` & `bfloat16` is currently unsupported on CPU but is being used in this test,
so I should've been more precise in that comment.

The CI check `pytorch_xla_linux_bionic_py3_6_clang9_test` would make it clear to me if `xla` also supports all operations used in this test for `float16` as well.
I think it does so now, as it seems to converts `Float16` to `float` prior to any operation.
Based on the CI check or your confirmation, I'll edit the remark regarding `xla`.


> Also, could we use @dtypes* instead of this?

Thanks, I'll refactor the test to do so.

Reply:
I took another look while refactoring and it's only `prod` that `float16` and `bfloat16` don't support  on CPU, so I enabled other checks in the test for these dtypes & now `float` is being used when either of the half-types are to be used for `prod`. Since something similar's being done for `int16` & `int32` as well, so please confirm if that's okay.